src/parser.ts

```diff
@@ -662,6 +672,60 @@ const TRAVERSE_BUILDER: MarkVisitor<(rhs: TraversalResult | null) => TraversalRe
   },
 }
 
+const SELECTOR_BUILDER: MarkVisitor<NodeTypes.SelectorNode> = {
+  this_attr(p) {
+    const name = p.processString()
+
+    if (['null', 'true', 'false'].includes(name)) throw new GroqSelectorError('invalid selector')
```

Comment:
Isn't this a bit naive, and ignores other invalid things like numbers? Should we not instead check that it is a valid identifier?